staging/src/k8s.io/apimachinery/pkg/runtime/lifecycle.go

```diff
@@ -0,0 +1,170 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package runtime
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/version"
+)
+
+var (
+	// DefaultAPILifecycleRegistry is the global registry of resource lifecycle information.
+	DefaultAPILifecycleRegistry APILifecycleRegistry = NewAPILifecycleRegistry()
```

Comment:
Our scheme is not global, why is a lifecycle registry global?

Reply:
the version a type is introduced or removed is deterministic (eventually by the annotation in types.go), independent of whether a specific scheme instance choose to include that resource or not.